var/spack/repos/builtin/packages/py-basemap/package.py

```diff
@@ -37,11 +37,23 @@ class PyBasemap(PythonPackage):
     # Per Github issue #3813, setuptools is required at runtime in order
     # to make mpl_toolkits a namespace package that can span multiple
     # directories (i.e., matplotlib and basemap)
-    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-setuptools', type='run')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-matplotlib', type=('build', 'run'))
     depends_on('pil', type=('build', 'run'))
     depends_on('geos')
 
+    def install(self, spec, prefix):
+        # The default `install` method adds `--single-version-external-managed`
+        # because the package depends on py-setuptools. But the package doesn't
+        # actually build with `py-setuptools`, it just needs it at run-time.
+        # Until I can figure out how to query direct build dependencies only,
+        # override the default `install` method.
```

Comment:
@adamjstewart: you can look in `setup.py`.  If it's imported unconditionally and actually used, I consider it one.  If it's conditionally imported (e.g. just for some options like building wheels) then you can look at how it's used to figure out if it's actually needed.  For most spack use cases it's not needed.